pulsar-broker/src/test/java/org/apache/pulsar/broker/resourcegroup/ResourceGroupConfigListenerTest.java

```diff
@@ -288,4 +298,27 @@ private void prepareData() throws PulsarAdminException {
         testAddRg.setDispatchRateInBytes(200L);
 
     }
+
+    @Test
+    public void testLoadAllResourceGroupsAsyncFailed() {
```

Comment:
I think you should add a test that responds to an error `Resource group does not exist: rg-1`, the same as the case you provided in the motivation.

Reply:
But for now implementation, you ignored the Exception when accepting a Notification of resource group changes, it would cause the same error you mentioned in Motivation, right?

Reply:
> This root cause loadAllResourceGroupsAsync is async call, we need to wait for this loading to complete, and then call new ResourceGroupNamespaceConfigListener.

You added 2 tests, and they only did the things below:
- call `new ResourceGroupNamespaceConfigListener`
- inject an Ex in the constructor of `ResourceGroupNamespaceConfigListener`

They are too simple to let others know why the error `Resource group does not exist: rg-1` throws, and why the current change can avoid the error.

I just know that this patch will make `new ResourceGroupNamespaceConfigListener` get an Ex immediately if `rgResources.listResourceGroupsAsync` throws an Ex, Pulsar ignored the Ex of `rgResources.listResourceGroupsAsync` before.